docs/SILDevelopment.md

```diff
@@ -0,0 +1,159 @@
+# SIL Development
+
+This document provides information for developers working on the
+implementation of SIL. The formal specification of the Swift
+Intermediate _Language_ is in SIL.rst. This document contains
+additional information specific to its implementation in the Swift
+compiler. Source comments normally provide this level of
+information. However, some features of the implementation are spread
+out across the source base. This documentation is meant to offer a
+central point for approaching the source base with links to the source
+for more detailed comments.
+
+## SILType
+
+TBD: Define the different levels of types. Explain type lowering with
+examples.
+
+## SILInstructionResults
+
+TBD: Explain how the various types fit together with pointers to the
+source: SILValue, SILInstruction, SingleValueInstruction,
+MultipleValueInstructionResult. And why it was done this way.
+
+## `SILFunction` and `apply` arguments.
+
+Throughout the compiler, integer indices are used to identify argument
+positions in several different contexts:
+
+- `SILFunctionType` parameters.
+
+- The SIL function definition's callee-side `SILFunctionArguments`,
+  including indirect results.
+
+- Caller-side "applied arguments" for `apply` and `try_apply` (vs. operands).
```

Comment:
What is the difference in between an "applied argument" vs an operand? That is not defined anywhere.